### PR TITLE
Remove device shim from tests

### PR DESCRIPTION
**Pull request by ray-milkey**

### CODE REVIEW

File: test/config/createremovetargettest.go [hunk 1]
```diff
@@ -39,22 +38,33 @@ const (
...
-	assert.NotNil(t, targetClient)
+	topoClient, err := gnmi.NewTopoClient()
+	assert.NotNil(t, topoClient)
 	assert.Nil(t, err)
-	timeout := 10 * time.Second
-	newTarget := &device.Device{
-		ID:      createRemoveTargetModDeviceName,
-		Address: createRemoveTargetModDeviceName + ":11161",
+
+	newDevice := &topo.Object{
```
Comment (adibrastegarnia) [inline test/config/createremovetargettest.go:45]: call it target

File: test/config/deletetest.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ func (s *TestSuite) TestDelete(t *testing.T) {
 	devices[0] = device1.Name()
 
 	// Wait for config to connect to the device
-	gnmi.WaitForTargetAvailable(t, device.ID(device1.Name()), 10*time.Second)
+	gnmi.WaitForTargetAvailable(t, topo.ID(device1.Name()), 10*time.Second)
```
Comment (adibrastegarnia) [inline test/config/deletetest.go:48]: same here target

Comment (ray-milkey) [reply test/config/deletetest.go:48]: That one is a package name

File: test/config/targetstatetest.go [hunk 3]
```diff
@@ -29,12 +28,21 @@ func (s *TestSuite) TestDeviceState(t *testing.T) {
...
+		assert.NoError(t, err)
+		var protocolStates []*topo.ProtocolState
+		if err != nil {
+			protocolStates = nil
+		} else {
+			protocolStates = protocols.State
+		}
+		return len(protocolStates) > 0 &&
+			protocolStates[0].Protocol == topo.Protocol_GNMI &&
+			protocolStates[0].ConnectivityState == topo.ConnectivityState_REACHABLE &&
```
Comment (adibrastegarnia) [inline test/config/targetstatetest.go:43]: We should not check these states anymore. The control relation defines connectivity. The entity can be in topo but not connected. I guess we should remove protocolState all together in tests.

File: test/config/transactiontest.go [hunk 4]
```diff
@@ -48,13 +48,13 @@ func (s *TestSuite) TestTransaction(t *testing.T) {
 	defer gnmi.DeleteSimulator(t, device1)
 
 	// Wait for config to connect to the devices
-	gnmi.WaitForTargetAvailable(t, device.ID(device1.Name()), time.Minute)
+	gnmi.WaitForTargetAvailable(t, topo.ID(device1.Name()), time.Minute)
```
Comment (adibrastegarnia) [inline test/config/transactiontest.go:51]: target here

File: test/config/transactiontest.go [hunk 5]
```diff
@@ -48,13 +48,13 @@ func (s *TestSuite) TestTransaction(t *testing.T) {
...
 
 	device2 := gnmi.CreateSimulator(t)
 	defer gnmi.DeleteSimulator(t, device2)
 
 	// Wait for config to connect to the devices
-	gnmi.WaitForTargetAvailable(t, device.ID(device2.Name()), time.Minute)
+	gnmi.WaitForTargetAvailable(t, topo.ID(device2.Name()), time.Minute)
```
Comment (adibrastegarnia) [inline test/config/transactiontest.go:57]: rename to target

File: test/ha/session_failover.go [hunk 6]
```diff
@@ -34,14 +33,27 @@ func (s *TestSuite) TestSessionFailOver(t *testing.T) {
...
+		protocols := &topo.Protocols{}
+		err = d.GetAspect(protocols)
+		assert.NoError(t, err)
+		var protocolStates []*topo.ProtocolState
+		if err != nil {
+			protocolStates = nil
+		} else {
+			protocolStates = protocols.State
+		}
+		return currentTerm == 1 && len(protocolStates) > 0 &&
```
Comment (adibrastegarnia) [inline test/ha/session_failover.go:52]: Same thing here about protocol state

File: test/config/transactiontest.go (hunk 4 above)
Comment (ray-milkey) [reply test/config/transactiontest.go:51]: package name

**Review by adibrastegarnia — APPROVED** [review]

**Pull request merged**